src/c_api.rs

```diff
@@ -391,12 +305,29 @@ fn json_api_get_type_internal<V: SelectValue>(v: &V) -> JSONType {
     }
 }
 
-pub fn json_api_get<M: Manager>(
-    _: M,
-    key: *mut c_void,
-    path: *const c_char,
-    jtype: *mut c_int,
-) -> *const c_void {
+pub fn json_api_next<M: Manager>(_: M, iter: *mut c_void) -> *const c_void {
+    let iter = unsafe { &mut *(iter as *mut ResultsIterator<M::V>) };
+    if iter.pos >= iter.results.len() {
+        null_mut()
+    } else {
+        let res = iter.results[iter.pos] as *const M::V as *const c_void;
+        iter.pos = iter.pos + 1;
+        res
+    }
+}
+
+pub fn json_api_len<M: Manager>(_: M, iter: *const c_void) -> c_ulong {
+    let iter = unsafe { &*(iter as *mut ResultsIterator<M::V>) };
+    iter.results.len() as c_ulong
```

Comment:
Rust `Vec` index is of type `usize`, user is passing `size_t`, such as in `json_api_get_at`
And now we also add `c_ulong`
Do we need it? Don't we want to stick to `libc` types?
(currently I see in libc
`pub type size_t = usize;`
But maybe this may change on other platforms)

Reply:
I thought size_t == usize ...